common/src/main/java/tachyon/TachyonURI.java

```diff
@@ -228,6 +232,43 @@ public TachyonURI getParent() {
   }
 
   /**
+   * Get the first n component of the TachyonURI path.
+   * There is no trailing separator as the path will be normalized by normalizePath() 
+   * 
+   * <pre>
+   * /a/b/c, 0              -> /
+   * /a/b/c, 1              -> /a
+   * /a/b/c, 2              -> /a/b
+   * /a/b/c, 3              -> /a/b/c
+   * /a/b/c, 4              -> null
+   * </pre>
+   * 
+   * @return the first n path component, null if the path has less than n components
+   */
+  public String getPathComponent(int n) {
```

Comment:
NIT: the current name `getPathComponent` is a bit confusing to me. How about `getLeadingPath`?